### PR TITLE
ARROW-14279: [Docs] Initial attempt at describing structure of PyArrow library

### PR DESCRIPTION
**Pull request by amol-**
Try to provide a description of how PyArrow is structured at high level so that people can understand where to find what and the general purpose of each file. The idea is to represent those based on general rules even if they don't apply 100% as exceptions shouldn't be exceptional enough to dictate the overall design

### CODE REVIEW

**Comment by amol-** [comment]
Any suggestion of what else can be documented to improve understanding of what's inside the ``pyarrow`` package is welcome

File: docs/source/python/getting_involved.rst [hunk 1]
```diff
@@ -33,3 +33,42 @@ integrations in other projects, we'd be happy to have you involved:
  * Follow our activity on `JIRA <https://issues.apache.org/jira/browse/ARROW>`_
  * Learn the `Format / Specification
    <https://github.com/apache/arrow/tree/master/format>`_
```
Comment (AlenkaF) [inline docs/source/python/getting_involved.rst:35]: Could we correct the indentation in the lines before the Architectural overview?

File: docs/source/python/getting_involved.rst [hunk 2]
```diff
@@ -33,3 +33,42 @@ integrations in other projects, we'd be happy to have you involved:
...
+--------------------
+
+PyArrow is for the major part a wrapper around the functionalities that
+Arrow C++ implementation provides. The library tries to take what's available
+in C++ and expose it through a user experience that is more pythonic and
+less complex to use. So while in some cases it might be easy to map what's
+in C++ to what's in Python, in many cases the C++ classes and methods are
+used as foundations to build easier to use entities.
+
+.. image:: py_arch_overview.svg
```
Comment (AlenkaF) [inline docs/source/python/getting_involved.rst:50]: The image is great and the content corresponding to it is very well structured and understandable!

I would also add alt text to make it more accessible for the readers. Fro example:

```suggestion
.. image:: py_arch_overview.svg
   :alt: Four layers of PyArrow architecture: .py, .pyx, .pxd and low level C++ code.
```

**Review by AlenkaF — APPROVED** [review]
Thank you for this content @amol- !

I only have two small suggestions, otherwise it is very clear and concise 👍

File: docs/source/python/getting_involved.rst (hunk 1 above)
Comment (amol-) [reply docs/source/python/getting_involved.rst]: done!

**Pull request closed without merging**